lib/parsers/iso_base_media_file_format/utils.rb

```diff
@@ -2,82 +2,86 @@
+        moov_box = box_tree.find { |box| box.type == 'moov' }
+        return unless moov_box
+        movie_matrix = moov_box.first_child('mvhd')&.fields&.[](:matrix) || IDENTITY_MATRIX
+        extreme_coordinates = video_trak_boxes(box_tree).each_with_object({}) do |trak_box, extreme_coordinates|
+          tkhd_box = trak_box.first_child('tkhd')
+          next unless tkhd_box
+          x = tkhd_box.fields[:width]
+          y = tkhd_box.fields[:height]
+          next unless x && y
+          track_matrix = tkhd_box.fields[:matrix] || IDENTITY_MATRIX
+          [[0, 0], [0, y], [x, 0], [x, y]].each do |coordinates|
+            x, y = (Matrix[[*coordinates, 1]] * track_matrix * movie_matrix).to_a[0][0..1]
+            extreme_coordinates[:min_x] = x if !extreme_coordinates[:min_x] || x < extreme_coordinates[:min_x]
+            extreme_coordinates[:max_x] = x if !extreme_coordinates[:max_x] || x > extreme_coordinates[:max_x]
+            extreme_coordinates[:min_y] = y if !extreme_coordinates[:min_y] || y < extreme_coordinates[:min_y]
+            extreme_coordinates[:max_y] = y if !extreme_coordinates[:max_y] || y > extreme_coordinates[:max_y]
+          end
+        end
+        unless extreme_coordinates.empty?
+          [
+            extreme_coordinates[:max_x] - extreme_coordinates[:min_x],
+            extreme_coordinates[:max_y] - extreme_coordinates[:min_y]
+          ]
+        end
       end
-    end
-    if extreme_coordinates.empty?
-      [nil, nil]
-    else
-      [
-        extreme_coordinates[:max_x] - extreme_coordinates[:min_x],
-        extreme_coordinates[:max_y] - extreme_coordinates[:min_y]
-      ]
-    end
-  end
 
-  def duration(box_tree)
-    mvhd_box = box_tree.find { |box| box.type == 'moov' }['mvhd']
-    mvhd_box[:duration] / mvhd_box[:timescale].to_f
-  end
+      def duration(box_tree)
+        mvhd_box = box_tree.find { |box| box.type == 'moov' }&.first_child('mvhd')
+        return unless mvhd_box
+        duration = mvhd_box.fields[:duration]
+        timescale = mvhd_box.fields[:timescale]&.to_f
+        duration / timescale if duration && timescale
+      end
 
-  def frame_rate(box_tree)
-    video_trak_boxes(box_tree).each_with_object([]) do |trak_box, frame_rates|
-      mdia_box = trak_box['mdia']
-      mdhd_box = mdia_box['mdhd']
-      stts_box = mdia_box['minf']['stbl']['stts']
+      def frame_rate(box_tree)
+        video_trak_boxes(box_tree).each do |trak_box|
+          mdhd_box = trak_box.first_descendent_by_path(%w[mdia mdhd])
+          stts_box = trak_box.first_descendent_by_path(%w[mdia minf stbl stts])
 
-      timescale = mdhd_box[:timescale].to_f
-      stts_entries = stts_box[:entries]
+          next unless mdhd_box && stts_box
 
-      frame_rate_for_entry = ->(stts_entry) { (timescale / stts_entry[:sample_delta]).truncate(2) }
+          timescale = mdhd_box.fields[:timescale]&.to_f
+          sample_delta = stts_box.fields[:entries]&.[](0)&.[](:sample_delta)
```

Comment:
I'm wondering, wouldn't something like this work?

```suggestion
          sample_delta = stts_box.fields.dig(:entries, 0, :sample_delta)
```

Reply:
Wasn't aware of dig - awesome!